apps/wallet/src/providers/pending-transactions-context.tsx

```diff
@@ -0,0 +1,112 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+
+import { getTransactionHash } from '@status-im/wallet/utils'
+
+import { Storage } from '../data/storage'
+
+import type { ApiOutput } from '@status-im/wallet/data'
+
+type Activity = ApiOutput['activities']['activities']['activities'][0]
+
+type PendingTransaction = Activity & {
+  status: 'pending'
+  uniqueId: string
+}
+
+type PendingTransactionsContext = {
+  pendingTransactions: PendingTransaction[]
+  addPendingTransaction: (
+    transaction: Omit<PendingTransaction, 'uniqueId'>,
+  ) => void
+  removePendingTransaction: (hash: string) => void
+  clearPendingTransactions: () => void
+  isLoading: boolean
+}
+
+const PendingTransactionsContext = createContext<
+  PendingTransactionsContext | undefined
+>(undefined)
+
+export function usePendingTransactions() {
+  const context = useContext(PendingTransactionsContext)
+  if (!context) {
+    throw new Error(
+      'usePendingTransactions must be used within PendingTransactionsProvider',
+    )
+  }
+  return context
+}
+
+export function PendingTransactionsProvider({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  const [pendingTransactions, setPendingTransactions] = useState<
+    PendingTransaction[]
+  >([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [storage] = useState(() => new Storage('pending-transactions'))
+
+  useEffect(() => {
+    const loadPendingTransactions = async () => {
+      try {
+        const result = await storage.get(['transactions'])
+        const stored = result.transactions as PendingTransaction[] | undefined
+
+        if (stored && Array.isArray(stored)) {
+          setPendingTransactions(stored)
+        }
+      } catch (error) {
+        console.error('Failed to load pending transactions:', error)
+      } finally {
+        setIsLoading(false)
+      }
+    }
+
+    loadPendingTransactions()
+  }, [storage])
+
+  useEffect(() => {
+    if (!isLoading) {
+      storage.set({ transactions: pendingTransactions }).catch(error => {
+        console.error('Failed to save pending transactions:', error)
+      })
+    }
+  }, [pendingTransactions, storage, isLoading])
```

Comment:
Non-blocking: Noticed that storage writes are triggered on every change to `pendingTransactions`. I think it works fine, but if we ever expect frequent updates, we might want to debounce it slightly to avoid excessive writes. Just something to keep in mind. No strong need unless we run into issues.